### PR TITLE
Add support for unboxed (default) values in Types

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
 Set non-nillable values to defaults in Types.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java [hunk 1]
```diff
@@ -267,6 +293,19 @@ private void renderUnionToSMethod(MemberShape memberShape) {
         }
     }
 
+    private class MemberDefaultVisitor extends ShapeVisitor.Default<String> {
+
+        @Override
+        protected String getDefault(Shape shape) {
```
Comment (mullermp) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java:299]: What about for Strings, blobs, etc? Is default the right visitor method?

File: codegen/smithy-ruby-codegen-test/model/component-test/main.smithy [hunk 2]
```diff
@@ -74,10 +75,34 @@ union Union {
...
 structure ClientError {
   Message: String
 }
 
 @error("server")
-structure ServerError {}
\ No newline at end of file
+structure ServerError {}
+
+operation DefaultsTest {
```
Comment (mullermp) [inline codegen/smithy-ruby-codegen-test/model/component-test/main.smithy:87]: This file is supposed to be for shared components. Can you move this out into another file?

File: codegen/smithy-ruby-codegen-test/integration-specs/types_spec.rb [hunk 3]
```diff
@@ -58,5 +58,21 @@ module Types
         expect(unknown_union.to_h).to eq({ unknown: { key: 'key', value: 'value' } })
       end
     end
+
+    describe DefaultsTestInput do
+      subject { DefaultsTestInput.new }
```
Comment (mullermp) [inline codegen/smithy-ruby-codegen-test/integration-specs/types_spec.rb:63]: I believe subject is implicitly defined already

**Review by mullermp — APPROVED** [review]

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java (hunk 1 above)
Comment (alextwoods) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java:299]: Currently yes - only the number types and String are supported.  We could implement all of the numbers individually here or just use the default.  Thoughts?

File: codegen/smithy-ruby-codegen-test/integration-specs/types_spec.rb (hunk 3 above)
Comment (alextwoods) [reply codegen/smithy-ruby-codegen-test/integration-specs/types_spec.rb]: Ah, yes it is.

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java (hunk 1 above)
Comment (mullermp) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TypesGenerator.java:299]: What's safest? If Blob is supported, then we're writing 0, right?

**Pull request merged**